platform/ext/target/nordic_nrf/common/core/native_drivers/spu.c

```diff
@@ -72,15 +73,65 @@ void spu_clear_events(void)
     nrf_spu_event_clear(NRF_SPU, NRF_SPU_EVENT_PERIPHACCERR);
 }
 
-void spu_regions_reset_all_secure(void)
+static bool spu_region_is_bootloader_region(NRF_SPU_Type * p_reg, uint8_t region_id)
+{
+    bool is_bootloader = false;
+
+#ifdef PM_MCUBOOT_ADDRESS
+    {
+        size_t start_id = (PM_MCUBOOT_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        size_t end_id = (PM_MCUBOOT_END_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        is_bootloader = is_bootloader || (region_id >= start_id && region_id <= end_id);
+    }
+#endif
+
+#ifdef PM_MCUBOOT_ADDRESS
+    {
+        size_t start_id = (PM_MCUBOOT_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        size_t end_id = (PM_MCUBOOT_END_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        is_bootloader = is_bootloader || (region_id >= start_id && region_id <= end_id);
+    }
+#endif
+
+#ifdef PM_B0_ADDRESS
+    {
+        size_t start_id = (PM_B0_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        size_t end_id = (PM_B0_END_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        is_bootloader = is_bootloader || (region_id >= start_id && region_id <= end_id);
+    }
+#endif
+
+#ifdef PM_S0_ADDRESS
+    {
+        size_t start_id = (PM_S0_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        size_t end_id = (PM_S0_END_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        is_bootloader = is_bootloader || (region_id >= start_id && region_id <= end_id);
+    }
+#endif
+
+#ifdef PM_S1_ADDRESS
+    {
+        size_t start_id = (PM_S1_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        size_t end_id = (PM_S1_END_ADDRESS - DEVICE_FLASH_BASE_ADDRESS) / FLASH_SECURE_ATTRIBUTION_REGION_SIZE;
+        is_bootloader = is_bootloader || (region_id >= start_id && region_id <= end_id);
+    }
+#endif
+
+    return is_bootloader;
```

Comment:
#ifdef PM_MCUBOOT_ADDRESS
is_bootloader |= some_static_function(region_id, PM_MCUBOOT_ADDRESS, PM_MCUBOOT_END_ADDRESS);
#endif

Let's reduce the amount of copy-paste to avoid copy-paste errors.